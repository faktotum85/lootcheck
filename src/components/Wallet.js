import React, { Component } from 'react';
import { connect } from 'react-redux';
import { deposit, withdraw } from '../actions/balance';

export class Wallet extends Component {
    state = {amount: undefined}

    updateAmount = event => {
        const amount = parseInt(event.target.value, 10);
        this.setState({amount});
    }

    deposit = () => {
        this.props.deposit(this.state.amount);
    }

    withdraw = () => {
        this.props.withdraw(this.state.amount);
    }

    render() {
        return (
            <div>
                <h3 className='balance'>Wallet balance: {this.props.balance}</h3>
                <br />
                <input type="text" className="input-wallet" onChange={this.updateAmount}/>
                <button className="btn-deposit" onClick={this.deposit}>Deposit</button>
                <button className="btn-withdraw" onClick={this.withdraw}>Withdraw</button>
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        balance: state.balance
    };
}

const mapDispatchToProps = dispatch => {
    return {
        deposit: amount => dispatch(deposit(amount)),
        withdraw: amount => dispatch(withdraw(amount))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Wallet);