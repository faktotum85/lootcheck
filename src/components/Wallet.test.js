import React from 'react';
import { shallow } from 'enzyme';
import Wallet from './Wallet';

describe('Wallet', () => {
    const store = {
        getState: () => 20,
        subscribe: () => {},
        dispatch: () => {}
    };
    const wallet = shallow(<Wallet store={store}/>).shallow();

    it('renders properly', () => {
        expect(wallet).toMatchSnapshot();
    });

    it('displays the balance from store', () => {
        console.log(wallet);
        expect(wallet.find('.balance').text()).toBe('Wallet balance: 20');
    });
});