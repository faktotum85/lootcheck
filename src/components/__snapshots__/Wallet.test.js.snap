// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={20}
    dispatch={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": undefined,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3
        className="balance"
>
        Wallet balance: 
        20
</h3>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Wallet balance: ",
          20,
        ],
        "className": "balance",
      },
      "ref": null,
      "rendered": Array [
        "Wallet balance: ",
        20,
      ],
      "type": "h3",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3
          className="balance"
>
          Wallet balance: 
          20
</h3>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wallet balance: ",
            20,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Wallet balance: ",
          20,
        ],
        "type": "h3",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
